src/lib/pubkey/sm2/sm2.cpp

```diff
@@ -68,34 +72,30 @@ std::vector<uint8_t> sm2_compute_za(HashFunction& hash,
    hash.update(get_byte<1>(uid_len));
    hash.update(user_id);
 
-   const size_t p_bytes = domain.get_p_bytes();
+   const size_t p_bytes = group.get_p_bytes();
 
-   hash.update(domain.get_a().serialize(p_bytes));
-   hash.update(domain.get_b().serialize(p_bytes));
-   hash.update(domain.get_g_x().serialize(p_bytes));
-   hash.update(domain.get_g_y().serialize(p_bytes));
+   hash.update(group.get_a().serialize(p_bytes));
+   hash.update(group.get_b().serialize(p_bytes));
+   hash.update(group.get_g_x().serialize(p_bytes));
+   hash.update(group.get_g_y().serialize(p_bytes));
```

Comment:
If you wanted to avoid a few heap allocations (not sure this is worth it, though):

```suggestion
   auto update_with_scalar = [bytes = std::vector<uint8_t>(group.get_p_bytes())](HashFunction& h,
                                                                                 const BigInt& scalar) mutable {
      scalar.serialize_to(bytes);
      h.update(bytes);
   };

   update_with_scalar(hash, group.get_a());
   update_with_scalar(hash, group.get_b());
   update_with_scalar(hash, group.get_g_x());
   update_with_scalar(hash, group.get_g_y());
```